docs/architecture/adr-028-change-pubkey.md

```diff
@@ -0,0 +1,81 @@
+# ADR 028: Change PubKey
+
+## Changelog
+
+- 30-09-2020: Initial Draft
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, in the Cosmos SDK, the address of an auth account is always based on the hash of the public key.  Once an account is created, the public key for the account is set in stone, and cannot be changed.  This can be a problem for users, as key rotation is a useful security practice, but is not possible currently.  Furthermore, as multisigs are a type of pubkey, once a multisig for an account is set, it can not be updated.  This is problematic, as multisigs are often used by organizations or companies, who may need to change their set of multisig signers for internal reasons.
+
+Transferring all the assets of an account to a new account with the updated pubkey is not sufficient, because some "engagements" of an account are not easily transferable.  For example, in staking, to transfer bonded Atoms, an account would have to unbond all delegations and wait the three week unbonding period.  Even more significantly, for validator operators, ownership over a validator is not transferrable at all, meaning that the operator key for a validator can never be updated, leading to poor operational security for validators. 
+
+## Decision
+
+We propose the creation of a new standaonle module called `changepubkey` that is an extension to `auth` that allows accounts to update the public key associated with their account, while keeping the address the same.
```

Comment:
What is the reasoning for having this in the comsos-sdk and not a separate repo?

Reply:
Like why is this in the Cosmos SDK repo?  I think this is a pretty important piece of auth that many chains would like to use.  It's also useful to have it in there, to ingrain the notion that the `address == hash(pubkey)` relationship does not always hold.

Reply:
I would want to be mindful of growing the maintenance overhead for the sdk maintainers. If we continue to add modules then the maintenance of these modules falls on them. If they are fine with it then it's okay.

Reply:
I designed this as a standalone module, but seems @aaronc prefers it be part of `x/auth` directly, so I think that's a good sign its core enough to the SDK

Reply:
Yes, it is a feature that has come up in some conversations and I have been thinking it belongs in `x/auth` along with a config flag. I have some further thoughts for how it could be useful to allow public key accounts to switch to group accounts in the future.